File: 0503-next-greater-element-ii/0503-next-greater-element-ii.js
var nextGreaterElements = function(nums) {
    const n = nums.length;
    const arr = new Array(n);
    for (let i = 0; i < n; i++) {
        let flag = true;
        for (let j = i + 1; j < n; j++) {
            if (nums[i] < nums[j]) {
                arr[i] = nums[j];
                flag = false;
                break;
            }
        }
        if (flag) {
            for (let j = 0; j < i; j++) {
                if (nums[i] < nums[j]) {
                    arr[i] = nums[j];
                    flag = false;
                    break;
                }
            }
        }
        if (flag) arr[i] = -1;
    }
    return arr;
};

File: 0503-next-greater-element-ii/0503-next-greater-element-ii.java
class Solution {
    public int[] nextGreaterElements(int[] nums) {
        int[] arr = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            boolean flag = false;
            for (int j = i + 1; j < nums.length; j++) {
                if (nums[i] < nums[j]) {
                    arr[i] = nums[j];
                    flag = true;
                    break;
                }
            }
            if (!flag) {
                for (int j = 0; j < i; j++) {
                    if (nums[i] < nums[j]) {
                        arr[i] = nums[j];
                        flag = true;
                        break;
                    }
                }
            }
            if (!flag) arr[i] = -1;
        }
        return arr;
    }
}